apps/onboarding/src/gateway/device-check/device-check.service.spec.ts

```diff
@@ -1,20 +1,31 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { DeviceCheckService } from './device-check.service';
 import { AppModule } from '../../app.module';
+import { HttpModule } from '@nestjs/common';
 
 describe('DeviceCheckService', () => {
   let service: DeviceCheckService;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      imports: [AppModule],
+      imports: [AppModule, HttpModule],
       providers: [DeviceCheckService],
     }).compile();
 
     service = module.get<DeviceCheckService>(DeviceCheckService);
   });
 
-  it('should be defined', () => {
-    expect(service).toBeDefined();
+  it('should be defineshould be true when the attestation is validd', async() => {
+    const result = true;
+    jest.spyOn(service, 'verifyDevice').mockImplementation(async () => result);
```

Comment:
This seems to just be testing the mock itself rather than any code in the service.